src/coreclr/vm/wasm/helpers.cpp

```diff
@@ -825,8 +574,11 @@ namespace
         if (table == nullptr)
         {
             StringToWasmSigThunkHash* newTable = new StringToWasmSigThunkHash();
-            for (const StringToWasmSigThunk& thunk : wasmThunks)
-                newTable->Add(thunk.key, thunk.value);
+            newTable->Reallocate(g_wasmThunksCount);
```

Comment:
```suggestion
            newTable->Reallocate(g_wasmThunksCount * StringToWasmSigThunkHash::s_density_factor_denominator / StringToWasmSigThunkHash::s_density_factor_numerator + 1);
```
This API has interesting contract... . It seems to expect the size of the internal storage.